eq_cir_management_ui/__init__.py

```diff
@@ -0,0 +1,126 @@
+"""Flask application factory for the EQ CIR Management UI."""
+
+import json
+import logging
+import os
+from pathlib import Path
+
+from flask import Flask
+from flask_talisman import Talisman
+from jinja2 import ChainableUndefined, ChoiceLoader, FileSystemLoader
+
+from eq_cir_management_ui.config.config import DefaultConfig
+from eq_cir_management_ui.errors.routes import errors_blueprint
+from eq_cir_management_ui.main.routes import main_blueprint
+from eq_cir_management_ui.utils.routes import utils_blueprint
+
+logger = logging.getLogger()
+
+talisman = Talisman()
+
+
+def create_app(app_config: type[DefaultConfig]) -> Flask:
+    """Flask application factory, used to isolate the instance of the Flask application.
+    See https://flask.palletsprojects.com/en/2.2.x/patterns/appfactories/ .
+    """
+    app = Flask(__name__)
+
+    app.config.from_object(app_config)
+    app.static_folder = str(Path("../static"))
+
+    app.register_blueprint(main_blueprint)
+    app.register_blueprint(errors_blueprint)
+    app.register_blueprint(utils_blueprint)
+
+    jinja_config(app)
+    design_system_config()
+    configure_secure_headers(app)
+
+    return app
+
+
+def env_override(value: str, key: str) -> str:
+    """Add a custom filter to Jinja to retrieve environment variables.
+
+    :param app: The Flask application.
+    """
+    return os.getenv(key, value)
+
+
+def jinja_config(app: Flask) -> None:
+    """Configuration for the Flask Jinja2 component. Here we provide a custom loader,
+    so we can load from an array of sources.
+
+    :param app: The Flask application.
+    """
+    # loader for local templates and design system component templates
+    file_system_loader = FileSystemLoader([Path("./node_modules/@ons/design-system"), Path("./templates")])
+
+    app.jinja_loader = ChoiceLoader([file_system_loader])
+    app.jinja_env.undefined = ChainableUndefined
+
+    # Clean up white space.
+    app.jinja_env.trim_blocks = True
+    app.jinja_env.lstrip_blocks = True
+
+    app.jinja_env.filters["env_override"] = env_override
+
+
+def design_system_config() -> None:
+    """Set the version of the design system to an environment variable and add an
+    environment variable filter so environment variables can be read from within
+    Jinja. This enables the design system version to be defined once within the
+    package.json file and then reused throughout the application. Primarily to
+    declare the CSS version to use.
+
+    :param app: The Flask application.
+    """
+    with open(Path("./package.json"), encoding="utf-8") as file:
+        package_json = json.load(file)
+        design_system_version = package_json["dependencies"]["@ons/design-system"]
```

Comment:
Directly accessing nested keys like ["dependencies"]["@ons/design-system"] without checking if they exist can lead to `KeyError`, this should be broken down (for readability as well).

Reply:
Have implemented via try/except, although Rhys is suggesting using the second method instead.
Will look again.

Reply:
Have re-implemented the second way